xinference/model/embedding/core.py

```diff
@@ -595,6 +595,53 @@ def encode(
 
             return all_embeddings, all_token_nums
 
+        @no_type_check
+        def _encode_clip(
+            model: SentenceTransformer,
+            sentences: Union[str, List[str]],
+            convert_to_numpy: bool = True,
+            **kwargs,
+        ):
+            import base64
+            import os
+            import re
+            from io import BytesIO
+
+            from PIL import Image
+
+            def base64_to_image(base64_str: str) -> Image.Image:
+                # base64_data = re.sub("^data:image/.+;base64,", "", base64_str)
+                base64_data = base64_str.split(",", 1)[1]
+                byte_data = base64.b64decode(base64_data)
+                image_data = BytesIO(byte_data)
+                img = Image.open(image_data)
+                return img
+
+            all_token_nums = 0
+            all_embeddings = []
+            texts = []
+            images = []
+            for obj in sentences:
+                if re.match(r"^data:image/.+;base64,", obj):
+                    image = base64_to_image(obj)
+                    images.append(image)
+                elif re.match(r"^https?://", obj) or os.path.exists(obj):
+                    images.append(obj)
+                else:
+                    texts.append(obj)
+                    all_token_nums += len(self._model.tokenize(obj))
+
+            # Encode texts and images
+            text_embeddings = model.encode(texts, normalize_embeddings=True)
```

Comment:
Can model encode the mixed texts and images?

If the users passed `[text, image, text, image]`, seems the encoded will be `[text_emb], [text_emb], [img_emb], [img_emb]`.

Reply:
Without `text` and `image` label, it is difficult to accept a mixture of texts and images. What do you think about compatibility with the [jina Embedding API](https://jina.ai/zh-CN/embeddings/)? 
```
curl https://api.jina.ai/v1/embeddings \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer jina_c5bea0ab8dbf4695a86e31394824f12144ZR3DNzArR0bXGm19kIOUs-jDCt" \
  -d @- <<EOFEOF
  {
    "model": "jina-clip-v2",
    "dimensions": 1024,
    "normalized": true,
    "embedding_type": "float",
    "input": [
        {
            "text": "A beautiful sunset over the beach"
        },
        {
            "text": "Un beau coucher de soleil sur la plage"
        },
        {
            "text": "海滩上美丽的日落"
        },
        {
            "text": "浜辺に沈む美しい夕日"
        },
        {
            "image": "https://i.ibb.co/nQNGqL0/beach1.jpg"
        },
        {
            "image": "https://i.ibb.co/r5w8hG8/beach2.jpg"
        }
    ]
  }
```

Reply:
Let me see why jina has this limitation.

That's an option also, I don't know if the `sentence` could be a dict, when it has key `text` or `image`, it will explicitly show the type.

Reply:
I think the order of text to image is just an example, I don't think it's a MUST.

I think we can allow dict be passed into sentences, if so, the `text` or `image` can be recognized.